line_beam_parameters.py

```diff
@@ -0,0 +1,138 @@
+    # to unchecked to start with
+    _bits = {'bit15': False, 'bit14': False, 'bit13': False, 'bit12': False,
+             'bit11': False, 'bit10': False, 'bit9': False, 'bit8': False,
+             'bit7': False, 'bit6': False, 'bit5': False, 'bit4': False,
+             'bit3': False, 'bit2': False, 'bit1': False, 'bit0': False}
+
+    # signal to emit when energy range is changed
+    energy_range_signal = QtCore.Signal(int)
+
+    # use this to help differentiate between changing
+    # the bits internally vs. externally
+    _bits_are_set = False
+    energy_channel = None
+
+    def __init__(self, parent=None, args=None, macros=None):
+        super(LineBeamParametersControl, self).__init__(parent=parent,
+                                                        args=args,
+                                                        macros=macros)
+        self.config = macros
+        self.setup_ui()
+
+        if self.energy_channel:
+            self.destroyed.connect(functools.partial(clear_channel,
+                                                     self.energy_channel))
+
+    def setup_ui(self):
+        self.setup_bits_connections()
+        self.setup_bit_indicators()
+        self.setup_energy_range_channel()
+
+    def setup_bits_connections(self):
+        """
+        Connect all the check boxes bits with the calc_energy_range method to
+        calculate the range upon changing a bit state.
+        """
+        for key, item in self._bits.items():
+            cb = self.findChild(QtWidgets.QCheckBox, f"{key}")
+            cb.stateChanged.connect(functools.partial(
+                self.calc_energy_range, key))
+
+    def setup_bit_indicators(self):
+        """
+        Borrowed function from fast_faults, to help morph the labels vertically
+        """
+        for key in self._bits.keys():
+            label = self.findChild(PyDMLabel, f"label_{key}")
+            if label is not None:
+                morph_into_vertical(label)
+
+    def calc_energy_range(self, key, state):
+        """
+        Catch when a check box is checked/unchecked and calculate
+        the current bitmask.
+
+        Parameters
+        ----------
+        key : str
+            The check box object name.
+        state : int
+            The state of the check box.
+            0 = unchecked
+            2 = checked
+
+        Note
+        ----
+        The checkboxes can be tri-states - here we use the states 0 and 2
+        for unchecked and checked respectively.
+        """
+        status = True if state == 2 else False
+        self._bits[key] = status
+        bit_map = list(map(int, [item for key, item in self._bits.items()]))
+        decimal_value = 0
+        for bit in bit_map:
+            # Shift left by pushing zeros in
+            # from the right and let the leftmost bits fall off
+            # Sets each bit to 1 if one of two bits is 1
+            decimal_value = (decimal_value << 1) | bit
```

Comment:
I really find neat that you used map here.

As a suggestion... 
You can go a step further and use `functools.reduce` to take care of calculating the new value.

```suggestion
decimal_value = functools.reduce(
    (lambda x, y: (x << 1) | y ),
    map(int, [item for key, item in _bits.items()])
)
```

Maybe it will be less readable... I don't know...

Reply:
I like this, thank you for the suggestion - I'm going to add it in.